File: 0134. Gas Station/Solution0134.py
from typing import List


class Solution0134:
    def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
        curgas, totgas, totcost, n, cand = 0, 0, 0, len(gas), 0
        for i in range(n):
            curgas += gas[i] - cost[i]
            if curgas < 0:
                curgas = 0
                cand = i + 1
            totgas += gas[i]
            totcost += cost[i]
        if totgas < totcost:
            return -1
        return cand
                

File: 0134. Gas Station/Solution0134.java
class Solution0134 { 
    public int canCompleteCircuit(int[] gas, int[] cost) {
        int curgas = 0;
        int totgas = 0, totcost = 0;
        int cand = 0;
        for (int i = 0; i < gas.length; i++) {
            curgas += gas[i];
            curgas -= cost[i];
            if (curgas < 0) {
                curgas = 0;
                cand = i+1;
            }
            totgas += gas[i];
            totcost += cost[i];
        }
        if (totgas < totcost) {
            return -1;
        }
        return cand;
    }
}